### ISSUE TITLE
TLS error on IoTDataPlane publish

### ISSUE THREAD

**Issue by michaelo-spindance** [aws/aws-sdk-js-v3#3305]
## Describe the bug

When running a node application in a docker container, `publish` requests using `IoTDataPlane` throw a TLS error. This behavior is consistent regardless if the container is running in ECS (our implementation is using Fargate) or running the container locally. When running the same code locally outside the container, no error is thrown.

#### SDK version number

`@aws-sdk/client-iot-data-plane@3.50.0`

#### Is the issue in the browser/Node.js/ReactNative?

Node.js

#### Details of the Node.js version

image: `node:16-alpine`
node: `v16.14.0`

## Steps to reproduce

`index.ts`

```typescript
import { IoTDataPlane } from '@aws-sdk/client-iot-data-plane';
const client = new IoTDataPlane({});
const topic = 'testing';
const payload = new TextEncoder().encode('hello world!');
client.publish({ topic, payload });
```

`Dockerfile`

```dockerfile
### Base
FROM public.ecr.aws/docker/library/node:16-alpine as base

RUN apk update --no-cache
WORKDIR /

# Copy source code
COPY . .

# Install prod dependencies
RUN apk add --no-cache --virtual .gyp libpq-dev python3 make g++ \
    && npm install --only=production \
    && apk del .gyp


### Builder
FROM base as builder

# Install build dependencies
RUN npm install

# Transpile typescript
RUN npm run build


### Runtime
FROM public.ecr.aws/docker/library/node:16-alpine as runtime
WORKDIR /usr/src/app

# Copy runtime dependencies
COPY --from=base ./node_modules ./node_modules
COPY --from=base ./package.json ./
COPY --from=builder ./dist ./

EXPOSE 3000
USER node
CMD ["node", "index.js"]
```

#### Observed behavior

The following error is thrown.

``` text
Error: unable to get local issuer certificate
--
at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
at TLSSocket.emit (node:events:520:28)
at TLSSocket._finishInit (node:_tls_wrap:944:8)
at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12) {
code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY',
'$metadata': { attempts: 1, totalRetryDelay: 0 }
}
```

#### Expected behavior

Successful publish to the MQTT topic.

## Additional context

The code snippet was also tested in a standalone EC2 instance that had used the same policy as the ECS task. The same error was thrown when executing `publish` using the node.js REPL.

As a sanity check, (to make sure that it wasn't a policy issue) the Python SDK was used to try to publish to the same topic. Here is the python code that was executed in the REPL:

```python
import boto3
iot = boto.client('iot-data')
iot.publish(topic='testing',payload=b'hello world')
```

The below response was received:

```json
{
    "ResponseMetadata": {
        "RequestId": "d588b474-6d7b-df7d-5673-6aaf56fdfb31",
        "HTTPStatusCode": 200, 
        "HTTPHeaders": {
            "content-type": "application/json", 
            "content-length": "65", 
            "date": "Thu, 10 Feb 2022 22:24:21 GMT",
            "x-amzn-requestid": "d588b474-6d7b-df7d-5673-6aaf56fdfb31",
            "connection": "keep-alive"
        }, 
    "RetryAttempts": 0
    }
}
```

**Comment by nbrinks** [comment]
Similar issue in the past on the Python SDK https://github.com/boto/boto3/issues/2686

**Comment by michaelo-spindance** [comment]
@yenfryherrerafeliz, the underlying issue turned out to be the default endpoint that was resolved for the `IoTDataPlane` client. The work around at the time was to explicitly request the ATS signed endpoint using the `IoT` client.

```typescript
import { IoT } from '@aws-sdk/client-iot';
import { IoTDataPlane } from '@aws-sdk/client-iot-data-plane';

const { endpointAddress } = await new IoT({}).describeEndpoint({ endpointType: 'iot:Data-ATS' });
const client = new IoTDataPlane({ endpoint: `https://${endpointAddress}` });
```

Sorry for not circling back to this issue. Looks like this was resolved with the [v3.60.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.60.0) release.

**Comment by yenfryherrerafeliz** [comment]
Hi @michaelo-spindance, thanks for letting me know. I will close this report, but if you happen to need anything that we can help you with, please feel free of opening a new issue.

Thanks!
